notebook/static/notebook/js/searchandreplace.js

```diff
@@ -188,20 +187,19 @@ define(function(require){
           $('<div/>').addClass('input-group')
           .append(
             $('<div/>').addClass('input-group-btn')
-              .append(isCaseSensitiveButton)
-              .append(isRegExpButton)
+                .append(isCaseSensitiveButton)
+                .append(isRegExpButton)
+                .append(onlySelectedButton)
           )
           .append(search)
         )
       )
       .append($('<div/>').addClass('form-group')
         .append(
-          $('<div/>').addClass('input-group')
-          .append(repl)
-          .append(
-            $('<div/>').addClass('input-group-btn')
-              .append(onlySelectedButton)
-          )
+          $('<div/>')
+              .addClass('input-group')
```

Comment:
see my PR to yours, which does what I was suggesting (though I may have not been clear) and seems to render fine.